common/Util-desktop.cpp

```diff
@@ -424,4 +425,57 @@ void alertAllUsers(const std::string&) {}
 /// No-op implementation in the test programs
 void alertAllUsers(const std::string&, const std::string&) {}
 #endif
+
+SysStopwatch::SysStopwatch()
+{
+    restart();
+}
+
+void SysStopwatch::restart()
+{
+    readTime(_startCPU, _startSys);
+}
+
+int SysStopwatch::getClockTicks()
+{
+#if defined __linux__
+    static int ticks = sysconf(_SC_CLK_TCK);
+    return ticks;
+#else
+    return 1000;
+#endif
 }
+
+void SysStopwatch::readTime(uint64_t &cpu, uint64_t &sys)
+{
+    cpu = 0;
+    sys = 0;
+#if defined __linux__
+    FILE* cg = fopen("/proc/self/stat", "r");
+    if (cg != nullptr)
+    {
+        char line[4096] = { 0 };
+        if (fgets(line, sizeof(line), cg))
+        {
+            StringVector bits = StringVector::tokenize(line, strlen(line), ' ');
+            cpu = std::atol(bits[13].c_str());
```

Comment:
Aha, man 5 proc explains these are the utime and stime fields, so this measures time, but only for this process, unlike measuring with std::chrono::system_clock::now().